awx/ui_next/src/components/ListHeader/ListHeader.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { Fragment } from 'react';
+import PropTypes, { arrayOf, shape, string, bool } from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import styled from 'styled-components';
+
+import DataListToolbar from '@components/DataListToolbar';
+import FilterTags from '@components/FilterTags';
+
+import {
+  encodeNonDefaultQueryString,
+  parseQueryString,
+  addParams,
+  removeParams
+} from '@util/qs';
+import { QSConfig } from '@types';
+
+const EmptyStateControlsWrapper = styled.div`
+  display: flex;
+  margin-top: 20px;
+  margin-right: 20px;
+  margin-bottom: 20px;
+  justify-content: flex-end;
+
+  & > :not(:first-child)  {
+    margin-left: 20px;
+  }
+`;
+class ListHeader extends React.Component {
+  constructor (props) {
+    super(props);
+
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleSort = this.handleSort.bind(this);
+    this.handleRemove = this.handleRemove.bind(this);
+    this.handleRemoveAll = this.handleRemoveAll.bind(this);
+  }
+
+  getSortOrder () {
+    const { qsConfig, location } = this.props;
+    const queryParams = parseQueryString(qsConfig, location.search);
+    if (queryParams.order_by && queryParams.order_by.startsWith('-')) {
+      return [queryParams.order_by.substr(1), 'descending'];
+    }
+    return [queryParams.order_by, 'ascending'];
+  }
+
+  handleSearch (key, value) {
```

Comment:
It would be nice if `handleSearch` and `handleRemove` had the same call signature - they both accept the same information (key, value) but one expects an object while the other expects them in an argument list.